contracts/core/modules/RebalancingSetCTokenIssuanceModule.sol

```diff
@@ -0,0 +1,593 @@
+/*
+    Copyright 2020 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+
+import { ReentrancyGuard } from "openzeppelin-solidity/contracts/utils/ReentrancyGuard.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { CommonMath } from "../../lib/CommonMath.sol";
+import { CompoundUtils } from "../../lib/CompoundUtils.sol";
+import { ICToken } from "../interfaces/ICToken.sol";
+import { IRebalancingSetToken } from "../interfaces/IRebalancingSetToken.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IWETH } from "../../lib/IWETH.sol";
+import { IAddressToAddressWhiteList } from "../interfaces/IAddressToAddressWhiteList.sol";
+import { ERC20Wrapper } from "../../lib/ERC20Wrapper.sol";
+import { RebalancingSetIssuanceModule } from "./RebalancingSetIssuanceModule.sol";
+
+
+/**
+ * @title RebalancingSetCTokenIssuanceModule
+ * @author Set Protocol
+ *
+ * A module that includes functions for issuing / redeeming rebalancing SetToken from its base components, cToken 
+ * underlying components, and Ether
+ */
+contract RebalancingSetCTokenIssuanceModule is
+    RebalancingSetIssuanceModule
+{
+    using SafeMath for uint256;
+
+    /* ============ State Variables ============ */
+
+    // Address and instance of AddressToAddressWhiteList contract
+    IAddressToAddressWhiteList public cTokenWhiteList;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Constructor function for RebalancingSetCTokenIssuanceModule
+     *
+     * @param _core                The address of Core
+     * @param _vault               The address of Vault
+     * @param _transferProxy       The address of TransferProxy
+     * @param _weth                Instance of Wrapped Ether
+     * @param _cTokenWhiteList     Instance of whitelisted cTokens to underlying addresses
+     */
+    constructor(
+        address _core,
+        address _vault,
+        address _transferProxy,
+        IWETH _weth,
+        IAddressToAddressWhiteList _cTokenWhiteList
+    )
+        public
+        RebalancingSetIssuanceModule(
+            _core,
+            _vault,
+            _transferProxy,
+            _weth
+        )
+    {
+        cTokenWhiteList = _cTokenWhiteList;
+
+        address[] memory cTokenAddresses = _cTokenWhiteList.validAddresses();
+
+        for (uint256 i = 0; i < cTokenAddresses.length; i++) {
```

Comment:
This is the 3rd time this logic has been replicated (cTokenExchangeIssuanceModule, cTokenIssuanceModule, and the bidder. Perhaps, we should just create a file that contract that is inherited that has 1) the cToken Whitelist state and the approval logic in the constructor.